src/meta/src/stream/stream_manager.rs

```diff
@@ -462,6 +455,7 @@ impl GlobalStreamManager {
             create_type,
             ddl_type,
             replace_table_job_info,
+            new_independent_compaction_group,
```

Comment:
Passing a boolean flag everywhere actually looks worse to me. 😕 I guess we can at least still keep it an `Options` struct, which can be made type-safer than `HashMap<String, String>` and allow us to extend it easily in the future.

Reply:
Yes, I want to add it later because the PR has been too large. I guess the `Option` will be `CreateStreamJobOption`?

Reply:
IMO the PR can not be any larger. 🤣 Just simply replace all changes of `pub new_independent_compaction_group: bool` to a struct. To me it seems strange to have this storage-specific option aside of other general options for a streaming job.